### PR TITLE
Add check to ensure the image does not have unacceptable vulnerabilities

### PR DESCRIPTION
**Pull request by samira-barouti**
podman-scap command output is used to count the number of vulnerabilities.

Please note that the generated HTML report is not being used to identify the number of vulnerabilities.

### CODE REVIEW

File: certification/internal/shell/has_minimal_vulns.go [hunk 1]
```diff
@@ -1,16 +1,133 @@
...
+
+	logger.Debugf("The number of found vulnerabilities: %d", numOfVulns)
+	if numOfVulns > 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func DownloadFile(filename string, url string) error {
```
Comment (skattoju) [inline certification/internal/shell/has_minimal_vulns.go]: could be factored into a "utils"

File: certification/internal/shell/has_minimal_vulns.go [hunk 2]
```diff
@@ -1,16 +1,133 @@
...
+		return err
+	}
+	defer out.Close()
+
+	// Write the body to file
+	_, err = io.Copy(out, resp.Body)
+	return err
 }
+
+func Unzip(bzipfile string, destination string) error {
```
Comment (skattoju) [inline certification/internal/shell/has_minimal_vulns.go]: could be factored into a "utils"

**Comment by tonytcampbell** [comment]
Would it be possible to add a unit test for this check?

**Comment by tonytcampbell** [comment]
Can we save the vuln.html content into a file on disk?  That way, if this check fails, the user will be able to look at the vuln.html file for important details and troubleshooting.

File: certification/internal/shell/has_minimal_vulns.go (hunk 1 above)
Comment (samira-barouti) [reply certification/internal/shell/has_minimal_vulns.go]: sounds like a great idea if I can ever make packages to work :/

Comment (samira-barouti) [reply certification/internal/shell/has_minimal_vulns.go]: Done! Please let me know if it looks good :)

Comment (skattoju) [reply certification/internal/shell/has_minimal_vulns.go]: nice looks good!

File: certification/internal/shell/has_minimal_vulns.go [hunk 3]
```diff
@@ -1,16 +1,89 @@
...
+
+	cmd := exec.Command("sudo", "oscap-podman", image, "oval", "eval", "--report", reportFile, ovalFilePathDecompressed)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+
+	if err != nil {
+		logger.Error("unable to execute oscap-podman on the image: ", cmd.Stderr)
```
Comment (skattoju) [inline certification/internal/shell/has_minimal_vulns.go:78]: could be a good idea to print the path to the report as well

File: certification/internal/shell/has_minimal_vulns.go [hunk 4]
```diff
@@ -1,16 +1,89 @@
...
+	}
+
+	numOfVulns, err := numberOfVulnerabilities(image, ovalFilePathDecompressed, logger)
+	if err != nil {
+		return false, err
+	}
+
+	logger.Debugf("The number of found vulnerabilities: %d", numOfVulns)
+	if numOfVulns > 0 {
+		return false, nil
```
Comment (skattoju) [inline certification/internal/shell/has_minimal_vulns.go:62]: Could be a good idea to print the path to the report if we find vulnerabilities. Not an issue now but might need to consider writing to something external when running as a job in a cluster.

Comment (samira-barouti) [reply certification/internal/shell/has_minimal_vulns.go:62]: very good idea! will do it

**Comment by bcrochet** [comment]
Has been moved to upstream repo. redhat-openshift-ecosystem/openshift-preflight#6

**Pull request closed without merging**